database/util.go

```diff
@@ -17,3 +18,22 @@ func GenerateAdvisoryLockId(databaseName string, additionalNames ...string) (str
 	sum = sum * uint32(advisoryLockIDSalt)
 	return fmt.Sprint(sum), nil
 }
+
+var errNoScheme = errors.New("no scheme")
+var errEmptyURL = errors.New("URL cannot be empty")
+
+// schemeFromURL returns the scheme from a URL string
+func SchemeFromURL(url string) (string, error) {
```

Comment:
Let's leave this method in the top-level util.go file since it's also used to parse schemes from source urls.

If you run into circular import issues, then we can create a new `internal/url` package and move this there.

Reply:
I moved it into its own package.